openvdb/math/Maps.cc

```diff
@@ -232,7 +232,8 @@ approxInverse(const Mat4d& mat4d)
         try {
             return mat4d.inverse();
         } catch (ArithmeticError& ) {
-            // Mat4 code couldn't invert.
+            OPENVDB_LOG_INFO(
```

Comment:
I'm concerned that this is going to generate misleading log messages.  A singular matrix is the expected input to the `approxInverse` function.

Reply:
I agree. I think the only viable alternative is to adjust the API here (with the appropriate deprecation warnings):

`bool approxInverse(Mat4d& result, const Mat4d& mat4d);`

And communicate failure through a return value rather than an exception? I see very minimal value in changing this code that has been in use for many years just to satisfy the requirements of static analysis, but it's really the only warning that's not trivial to fix so we should probably just do it.

Reply:
`approxInverse` doesn't communicate failure through an exception, at least not in the case of a singular input matrix.  It's swallowing the exception, not throwing one.

If you're proposing instead to change the `Mat::inverse` API just to silence this static analyzer, I'm opposed to that.

Reply:
Yes, oops, that's what I meant. A change to the API such as:

`bool inverse(Mat4& result, T tolerance = 0) const;`
`Mat4 inverse(T tolerance = 0, bool& result) const;`

What's your opposition? Sounds like you think we should just not worry about fixing this particular warning?

Reply:
Changing the API would mean that `inverse` could no longer be used within expressions ("`b = a.inverse() * x * a`") and/or that failure would be too easily ignored.

Is not fixing any of these warnings an option?  I mean, why is "`catch (...) { status = false; }`" any more secure than "`catch (...) {}; status = false;`"?

Reply:
I totally agree. I'm going to close this PR. This is only a warning after all, if it was easy to fix we